src/lib/rsa/sandboxed/RSAKeysIframe.js

```diff
@@ -0,0 +1,85 @@
+/* global forge */
+
+/**
+ * @typedef {{command: 'generateKey', seed: string, keySize: number}} GenerateKeyCommand
+ * ...add new command types here and add them to the `Command` union type below
+ *
+ * @typedef {GenerateKeyCommand} Command
+ */
+
+const $loading = /** @type {HTMLSpanElement} */ (document.querySelector('#loading'));
+
+/**
+ * @param {any} key
+ * @returns {ArrayBuffer}
+ */
+function key2ArrayBuffer(key) {
+    const bytes = /** @type {string} */ (forge.asn1.toDer(key).getBytes());
+    const buffer = new ArrayBuffer(bytes.length);
+    const view = new Uint8Array(buffer);
+    for (let i = 0; i < view.length; i++) {
+        view[i] = bytes.charCodeAt(i);
+    }
+    return buffer;
+}
+
+window.addEventListener('message', async event => {
+    if (event.source === event.target) {
+        // console.log("Ignored same-window event:", event);
+        return;
+    }
```

Comment:
Messages which are not coming from the Keyguard should be rejected.
```suggestion
    if (event.origin !== window.location.origin || !event.source) return; // Reject messages from other origins
```

Reply:
Resolved in #524